pandas/_libs/tslibs/offsets.pyx

```diff
@@ -310,6 +310,16 @@ cdef _determine_offset(kwds):
                                'year', 'month', 'week', 'day', 'weekday',
                                'hour', 'minute', 'second', 'microsecond')
 
+    if any(not k.endswith('s') for k in kwds_no_nanos):
+        if 'weekday' not in kwds_no_nanos:
+            warnings.warn(
+                "The DateOffset arguments that do not end in 's' replace "
```

Comment:
I'd write `Use of Date Offset arguments that do not end with 's' overwrites the offset instead of incrementing it. These arguments are deprecated now and slated for removal in a future version.`

Reply:
Looks better. I will make the change.